MaxText/layers/attentions.py

```diff
@@ -961,11 +961,15 @@ def kv_projection(self, inputs_kv: Array, proj_name: str) -> Array:
     if self.num_query_heads % self.num_kv_heads != 0:
       raise ValueError("Invalid num_kv_heads for GQA.")
 
+    kernel_axes = ("embed", "heads", "kv")
+    if self.config.model_name == "llama2-70b" and self.config.sharding_strategy == "experimental":
```

Comment:
Can we use the more specific (embed, kv_heads, kv_head_dim) for all models, and instead its up to the config to decide how to shard this specific dim (e.g. add the default sharding in base.yml for kv_heads that makes this a no-op), but you can have wahtever you want in llama2-70b. I'd prefer this over the "experimental" or model specific if statements - ideally everything can be specified in the config and use the same generic code

Reply:
Same as short term vs long term.

Reply:
+1, can you add  `kv_proj_kernel_axes` to config and specialize in llama-70-v5e-16 config? Adding model specific logic in the core attention logic is the the preferred way.